src/3d.js

```diff
@@ -123,6 +223,84 @@ export var hackOptions = {
 	// but if you have a big, highly branching game with lots of art,
 	// you may want to disable it
 	preloadTextures: true,
+
+    // function used to adjust mesh instances after they have been added to the scene
+	meshExtraSetup: function (drawing, mesh) {
+        var name = drawing.name || '';
+
+        // transform tags. #m(x,y,z): move, #r(x,y,z): rotate, #s(x,y,z): scale
+        // #m(1,0,0.5) and #m(1,,.5) are both examples of valid input
+        // scale
+        var scaleTag = name.match(/#s\((-?[\.?\d+]\.?\d*)?,(-?[\.?\d+]\.?\d*)?,(-?[\.?\d+]\.?\d*)?\)/);
+        if (scaleTag) {
+            mesh.scaling = new BABYLON.Vector3(Number(scaleTag[1]) || 1,
```

Comment:
i'm not sure... i tested it in the browser console and it appears that '||' works as expected here.. i made it 1 for scaling because i thought it would make more sense as a default value for scaling if you scale by 1, nothing changes, but when you scale by 0 it does. isn't || supposed to evaluate the left side first? so if it is not undefined or null it would just return it and otherwise return whatever is on the rigth side?

Reply:
so specifically the way `||` works is that it's the "or" operator; you're correct about it evaluating the left side first, and evaluating + returning the right side only if it didn't use the left, but because it's a boolean operator, it makes the decision based on what the value would be if cast to a boolean, i.e. "truthy" and "falsey" values. `undefined` and `null` are both falsey, but `0` is also falsey, so if you were to do something like `#s(0,1,1)`, it would be interpreted here as `#s(1,1,1)`